bcda/router.go

```diff
@@ -29,13 +29,16 @@ func NewAPIRouter() http.Handler {
 		r.With(auth.RequireTokenAuth, auth.RequireTokenJobMatch).Get(m.WrapHandler("/jobs/{jobID}", jobStatus))
 		r.Get(m.WrapHandler("/metadata", metadata))
 	})
-	r.Post(m.WrapHandler("/auth/token", getAuthToken))
 	r.Get(m.WrapHandler("/_version", getVersion))
 	r.Get(m.WrapHandler("/_health", healthCheck))
 	r.Get(m.WrapHandler("/_auth", getAuthInfo))
 	return r
 }
 
+func NewAuthRouter() http.Handler {
+	return auth.NewAuthRouter(auth.ParseToken, logging.NewStructuredLogger(), HSTSHeader, ConnectionClose)
```

Comment:
> ParseToken is for external consumption by users of auth, because the token claims it expects are for the API. Token claims for its auth admin endpoints like group would be different. ParseToken should not be middleware for this call.

@DeirdreHolub Is this just a semantic issue, since the parameters are named "middleware" in `auth.NewAuthRouter`?  The API is still in full control over whether `ParseToken` is called.  It's used in exactly this way 3 lines up in a call to `chi.With()` (which also names these parameters as middleware).  The one change is that it's now being passed to `auth.NewAuthRouter`.

From a risk perspective I'm less worried about whether we're now hiding `ParseToken` inside auth, and more worried whether there are other customizations to an `http.Handler` that might need to be made in other contexts or by other clients.

Reply:
ParseToken removed.  Thanks for clarifying!